invenio_records_resources/services/records/queryparser/transformer.py

```diff
@@ -33,15 +33,57 @@ def term_name(self):
         """Get the term name."""
         return self._term_name
 
-    def map_word(self, node):
+    def map_word(self, node, **kwargs):
         """Modify a word node."""
         return self._word_fun(node) if self._word_fun else node
 
-    def map_phrase(self, node):
+    def map_phrase(self, node, **kwargs):
         """Modify a phrase node."""
         return self._phrase_fun(node) if self._phrase_fun else node
 
 
+class RestrictedTerm:
+    """Class used to apply specific permissions to search."""
+
+    def __init__(self, permission, term_name=None, word=None, phrase=None):
+        """Constructor."""
+        self.permission = permission
+        self._word_fun = word
+        self._phrase_fun = phrase
```

Comment:
```suggestion
        self._word_func = word
        self._phrase_func = phrase
```
and I would change the above too